server/controllers/dashboardController.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+import moment from "moment";
+import * as util from "util";
+import models from "../models";
+import { Op } from "sequelize";
+import paginationMeta from "../helpers/paginationHelper";
+import {paginationResponse} from "../utils/formatter";
+import {validFormat} from '../helpers/dateHelpers';
+
+
+const automation = models.Automation;
+/**
+ * Returns dateFunction
+ * @param {object} date - date string
+ * @returns {object} - date object
+ */
+const dateFunction = (date = { endDate: moment(), startDate: moment().subtract(30, 'days')}) => {  
+  // eslint-disable-next-line no-unused-vars2
+  const endDate = moment(date.endDate).format("YYYY-MM-DD");
+  const startDate = moment(date.startDate).format("YYYY-MM-DD");
+  // check if date.from is greater than date.to or today, return an error
+  if (startDate > endDate) {
+    throw new Error("startDate cannot be greater than endDate");
+  }
+  // check if both date from and to are provided
+  if (startDate && endDate) {
+    return { dateFrom: startDate, dateTo: endDate };
+  }
+};
+
+/**
+ * Returns dateFunction
+ * @param {string} type - boarding type
+ * @param {number} offset - start value
+ * @param {number} limit - number of records to pick
+ * @param {Date} dateFrom - start date
+ * @param {Date} dateTo - end date
+ * @returns {object} - upselling partner record
+ */
+ const UpsellingPartnerQuery = async (type, offset, limit, dateFrom, dateTo) => {
+  const val = await automation
+  .findAndCountAll({
+    attributes: [
+      [models.sequelize.fn("COUNT", models.sequelize.col("*")), "count"],
+      "type",
+      "partnerName"
+    ],
+    where: {
+      createdAt: {
+        [Op.between]: [
+          dateFrom,
+          dateTo
+        ]
+      },
+      type: type
+    },
+    offset: offset,
```

Comment:
Since the key and value are same, we don't need to do:
```
{
   offset: offset,
   limit: limit,
}
```
We can do instead:
```
{
   offset,
   limit,
}
```

Reply:
Done